### PR TITLE
[AIRFLOW-5458] Bump Flask-AppBuilder to 2.2.0

### PR DESCRIPTION
**Pull request by paulvic**
Same fix as was required for Superset (https://github.com/apache/incubator-superset/issues/7739)

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5458
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
When using OAuth for authentication, if the `next` request parameter isn't set in the URL then the initial redirect is broken after logging in. This was fixed in FAB (https://github.com/dpgaspar/Flask-AppBuilder/pull/1095) and so the version of FAB needs to be bumped to pick up the fix.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
I'm not sure if we would add tests for an external package.

I have tested this manually and confirmed the issue is resolved.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by ashb** [comment]
All in favour of this change, but have you checked the rest of the UI? Has anything else around filtering or searching broken or changed?

**Comment by paulvic** [comment]
> All in favour of this change, but have you checked the rest of the UI? Has anything else around filtering or searching broken or changed?

I logged in with two different accounts, one with `admin` and the other with `user` role. Verified the admin had the typical admin permissions. Verified the user had limited permissions and searching, triggering DAGs, viewing logs, etc. still works correctly.

**Comment by feng-tao** [comment]
how about a custom DAG user role?

**Comment by feng-tao** [comment]
cc @mistercrunch  Max, do you know if FAB changes any API/interface that Airflow is using?

**Comment by zachliu** [comment]
We've been experiencing a similar issue as AIRFLOW-5462. Everything works as expected for a valid login using the company domain specified in `OAUTH_PROVIDERS`'s `whitelist`. However, if I try to login with, say, a personal Gmail account, it breaks the UI with same error traceback as AIRFLOW-5462. And unfortunately this PR doesn't solve this issue :sob:

File: setup.py [hunk 1]
```diff
@@ -365,7 +365,7 @@ def do_setup():
             'cryptography>=0.9.3',
             'dill>=0.2.2, <0.4',
             'flask>=1.1.0, <2.0',
-            'flask-appbuilder>=1.12.5, <2.0.0',
+            "flask-appbuilder>=2.2.0, <2.3.0",
```
Comment (ashb) [inline setup.py]:
```suggestion
            'flask-appbuilder>=2.2.0, <2.3.0',
```

Comment (ashb) [reply setup.py]: Also is there a reson we should limit it to 2.3, Would `<3`, or `~=2.2` be better? https://www.python.org/dev/peps/pep-0440/#compatible-release

Comment (ashb) [inline setup.py]:
```suggestion
            "flask-appbuilder~=2.2",
```

Comment (feluelle) [reply setup.py]: I would suggest doing it like this `"flask-appbuilder~=2.2",`

**Review by feluelle — APPROVED** [review]
I cherry-picked this line to v1-10-test ( 5c3813bd745d88d713972b23f60cd5eecacb0ce9 )

Can confirm that UI works with `Flask-AppBuilder               2.2.1`

**Comment by zachliu** [comment]
I'm still having the same error using `flask-AppBuilder 2.2.1` :sob:

**Comment by ashb** [comment]
@zachliu Yeah, it sounds like this change won't fix the OAuth issue :( , but will improve a few other things :)

File: setup.py (hunk 1 above)
Comment (kaxil) [inline setup.py]:
```suggestion
            'flask-appbuilder~=2.2,
```

**Comment by kaxil** [comment]
@paulvic Can you update the PR based on the suggestion and we can use the same PR to resolve https://issues.apache.org/jira/browse/AIRFLOW-5458, we want to do this for 1.10.7

**Comment by ashb** [comment]
Fab 2.0.0 only works on py3.6+ - on the release branch this change has been changed so it's this:

```
            'flask-appbuilder>=1.12.2, <=2.0.0;python_version<"3.6"',
            'flask-appbuilder~=2.2;python_version>="3.6"',
```

**Pull request merged**